src/referral/referral_utils.cairo

```diff
@@ -31,6 +31,11 @@ fn set_trader_referral_code(
     if (referral_code == 0) {
         return;
     }
+    // skip setting of the referral code if the user already has a referral code
+    if (referral_storage.trader_referral_codes(account) != 0) {
+        return;
+    }
```

Comment:
I think we should allow changing the referral code

Reply:
ok. I will fix it

Reply:
updated here
https://github.com/tevrat-aksoy/satoru/commit/b844700e96c70fa80a4ab35e3f6d701b592f9c7d